apps/meteor/app/livechat/server/api/v1/room.ts

```diff
@@ -98,6 +98,10 @@ API.v1.addRoute(
 		async post() {
 			const { rid, token } = this.bodyParams;
 
+			if (!rcSettings.get('Omnichannel_allow_visitors_to_close_conversation')) {
+				throw new Error('error-not-allowed-to-close-conversation');
```

Comment:
I'm thinking, shouldn't we return `API.v1.unauthorized()` instead? I believe this fits more the idea than a regular error (which will return code 400), but I'd like to know what you think

Reply:
Not sure, technically the visitor doesn't have any permission attached. And they never log in so for them there's no such thing as being authorized or not...

But since the wording says "not allowed" then it may be confusing 🤔 

So, not sure lol. I guess no other endpoint returs 401/403 for any visitor action so im inclined to leave the 400

Reply:
Shouldn't we just hide/not render the close conversation button from the user if they can't close the conversation?

Also, I think 403 forbidden is a more appropriate status code.

Reply:
Yep, thats exactly on the other changes ;)

Again, I'm really not sure. We don't throw any 401/403 errors on these visitor endpoints because technically they're not authenticated, and this is not a permission of them, it's a configuration.

Reply:
(We also have a test of the button not appearing btw)

Reply:
Sorry! I made the comment before noticing those changes and forgot to update the comment.

Reply:
As for the status code.

403 would be the more appropriate status for this, but... if we have a standard of using the 400 instead, we should keep to it to avoid confusing, or change all of them to use the appropriate status code... but that is another subject. haha

Reply:
@MarcosSpessatto would like to hear your opinion about this
I think Kevin has a good point, but we do use `unauthorized()` for similar use cases out of the livechat context. So I'm not sure

Reply:
Can you share some samples? 🤗 if we have enough maybe we can start with the new pattern, otherwise I'd think it's even better to change them lol.

Reply:
I agree with the pattern, even if 403 looks nicer and more appropriate, I agree with follow the pattern we already have